pinning-services/index.js

```diff
@@ -81,73 +94,126 @@ const job = new cron.CronJob(period, function() {
         const dirWww = `${projDir}/${projName}/www`;
         const dirApi = `${projDir}/${projName}/api`;
 
+        // Read publication settings from project configuration
+        let protoHttp = true;
+        let protoHttpPurgeIfDisabled = true;
+        let protoIpfs = true;
+        let protoIpfsPurgeIfDisabled = true;
+        let protoHypercore = true;
+        let protoHypercorePurgeIfDisabled = true;
+        if (proj['publication'] && proj['publication']['protocol']) {
+
+          // Check for known protocols and set flags
+          Object.keys(proj['publication']['protocol']).forEach((item) => {
+            switch (item) {
+              case 'http':
+                protoHttp = proj['publication']['protocol'][item]['enabled'] === true;
+                protoHttpPurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] === true
+                break;
+              case 'ipfs':
+                protoIpfs = proj['publication']['protocol'][item]['enabled'] === true;
+                protoIpfsPurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] === true
+                break;
+              case 'hypercore':
+                protoHypercore = proj['publication']['protocol'][item]['enabled'] === true;
+                protoHypercorePurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] === true
+                break;
```

Comment:
I think if the input is wrong type, it should default to the default value, which is `true` for these:

```suggestion
              case 'http':
                protoHttp = proj['publication']['protocol'][item]['enabled'] !== false;
                protoHttpPurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] !== false;
                break;
              case 'ipfs':
                protoIpfs = proj['publication']['protocol'][item]['enabled'] !== false;
                protoIpfsPurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] !== false;
                break;
              case 'hypercore':
                protoHypercore = proj['publication']['protocol'][item]['enabled'] !== false;
                protoHypercorePurgeIfDisabled = proj['publication']['protocol'][item]['purgeIfDisabled'] !== false;
                break;
```